server/controllers/finance/purchase.js

```diff
@@ -3,152 +3,213 @@ var db = require('./../../lib/db');
+      'FROM purchase ' +
+      'JOIN creditor ON creditor.uuid = purchase.creditor_uuid ' +
+      'JOIN employee ON employee.id = purchase.purchaser_id ' +
+      'JOIN user ON user.id = purchase.emitter_id; ';
+  }
+
+  db.exec(sql)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
 
-  params = values(data);
-  params.push(primaryCashId);
-  params.push(userId);
+function detail (req, res, next) {
+  'use strict';
 
-  return db.exec(sql, params);
+  var purchaseUuid = req.params.uuid,
+    purchaseRow, purchaseItems;
+
+  var sqlPurchase =
+    'SELECT purchase.uuid, purchase.reference, purchase.cost, purchase.discount, purchase.purchase_date, purchase.paid, ' +
+    'creditor.text, employee.name, employee.prenom, user.first, user.last ' +
+    'FROM purchase ' +
+    'JOIN creditor ON creditor.uuid = purchase.creditor_uuid ' +
+    'JOIN employee ON employee.id = purchase.purchaser_id ' +
+    'JOIN user ON user.id = purchase.emitter_id ' +
+    'WHERE purchase.uuid = ?';
+
+  var sqlPurchaseItem =
+    'SELECT purchase_item.purchase_uuid, purchase_item.uuid, purchase_item.quantity, purchase_item.unit_price, ' +
+    'purchase_item.total, inventory.text ' +
+    'FROM purchase_item ' +
+    'JOIN inventory ON inventory.uuid = purchase_item.inventory_uuid ' +
+    'WHERE purchase_item.purchase_uuid = ? ';  
+
+  db.exec(sqlPurchase, [purchaseUuid])
+  .then(function (row) {
+    purchaseRow = row;
+    return db.exec(sqlPurchaseItem, [purchaseUuid]);
+  })
+  .then(function (rows) {     
+    if (purchaseRow.length === 0) { 
+      res.status(404).json({
+        code : 'ERR_NOT_FOUND', 
+        reason : 'No Purchase found under the uuid ' + purchaseUuid
+      });
+    } else {       
+      // Found Purchase resource - unpack and return to client
+      purchaseItems = rows;
+      res.status(200).json({
+        purchase : purchaseRow, 
+        purchaseItems : purchaseItems
+      });
+    }
+  })  
+  .catch(next)
+  .done();
 }
 
-function writeCashItems(primaryCashId, data) {
-  var sql, queries, params;
+
+// PUT /purchase/:uuid
+function update(req, res, next) {
+  'use strict';
+
+  var sql;
 
   sql =
-    'INSERT INTO `primary_cash_item` ' +
-      '(inv_po_id, debit, credit, document_uuid, uuid, primary_cash_uuid) ' +
-    'VALUES ' +
-      '(?, ?, ?, ?, ?, ?)';
-
-  queries = data.map(function (item) {
-    params = values(item);
-    params.push(uuid());
-    params.push(primaryCashId);
-    return db.exec(sql, params);
-  });
+    'UPDATE purchase SET ? WHERE uuid = ?;';
 
-  return q.all(queries);
-}
+  db.exec(sql, [req.body, req.params.uuid])
+  .then(function () {
 
-function postToJournal(primaryCashId, userId) {
-  var dfd = q.defer();
-  journal.request('indirect_purchase', primaryCashId, userId, function (err, result) {
-    if (err) { return dfd.reject(err); }
-    dfd.resolve(result);
-  });
-  return dfd.promise;
+    sql =
```

Comment:
You can then use `lookupPurchaseOrder()` from both the `detail()` and `update()` methods.  [Here's an example from the cash payment module](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/controllers/finance/cash.js#L20)